src/Rules/Variable/AbstractEndNameRule.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules\Variable;
+
+use TwigCsFixer\Rules\AbstractFixableRule;
+use TwigCsFixer\Token\Token;
+use TwigCsFixer\Token\Tokens;
+
+/**
+ * Ensures that the name is set at the end.
+ */
+abstract class AbstractEndNameRule extends AbstractFixableRule
+{
+    protected function process(int $tokenIndex, Tokens $tokens): void
+    {
+        $token = $tokens->get($tokenIndex);
+        if (!$token->isMatching(Token::BLOCK_NAME_TYPE, $this->getEndName())) {
+            return;
+        }
+
+        $nameToken = $this->findNameToken($tokenIndex + 1, $tokens);
+        if ($nameToken instanceof Token) {
+            return;
+        }
+
+        $indent = 0;
+        $index = $tokenIndex - 1;
+        while ($tokens->has($index)) {
+            $current = $tokens->get($index);
+            if ($current->isMatching(Token::BLOCK_NAME_TYPE, $this->getEndName())) {
+                ++$indent;
+                --$index;
+                continue;
+            }
+
+            if ($current->isMatching(Token::BLOCK_NAME_TYPE, $this->getStartName())) {
+                if ($indent > 0) {
+                    --$indent;
+                } elseif (0 === $indent) {
+                    break;
+                }
+            }
+            --$index;
+        }
+
+        $nameToken = $this->findNameToken($index + 1, $tokens);
```

Comment:
You should rely on something like
```
$tokens->findNext([Token::NAME_TYPE, Token::MACRO_NAME_TYPE], $index + 1);
```
instead.

One tricks can be to do
```
$tokens->findNext([Token::NAME_TYPE, Token::MACRO_NAME_TYPE, BLOCK_END_TYPE], $index + 1);
```
and if the match is `BLOCK_END_TYPE` you know there is no name.